### PR TITLE
Infer the module name from the go.mod file

### PR DESCRIPTION
**Pull request by dprotaso**
Fixes: https://github.com/knative/actions/issues/28

### CODE REVIEW

**Comment by dprotaso** [comment]
Testing here - so it's ready https://github.com/knative/pkg/pull/2509

File: go/downstream-test/action.yaml [hunk 1]
```diff
@@ -1,45 +1,43 @@
...
-  downstream-module:
-    description: 'Downstream Module. For example, knative.dev/sample-controller'
     required: true
 
   downstream-path:
     description: 'Path to the downstream module - defaults to the module name'
-    required: false
-
+    required: true
+
```
Comment (kvmware) [inline go/downstream-test/action.yaml:11]: WS

File: go/downstream-test/README.md [hunk 2]
```diff
@@ -33,14 +33,14 @@ steps:
...
     uses: actions/checkout@v3
     with:
       repository: knative-sandbox/sample-controller
-      path: knative.dev/sample-controller
+      path: controller
   - uses: knative/actions/go/downstream-test@main
     with:
-      upstream-module: knative.dev/pkg
-      downstream-module: knative.dev/sample-controller
+      upstream-path: pkg
```
Comment (kvmware) [inline go/downstream-test/README.md:44]: The readme gives the example `./knative.dev/pkg` but only `pkg` is here?

Comment (dprotaso) [reply go/downstream-test/README.md:44]: let me add an update

**Pull request merged**